pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -42,58 +44,72 @@ import (
 	samplescheme "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned/scheme"
 	informers "github.com/xmudrii/etcdproxy-controller/pkg/client/informers/externalversions/etcd/v1alpha1"
 	listers "github.com/xmudrii/etcdproxy-controller/pkg/client/listers/etcd/v1alpha1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-const controllerAgentName = "etcdstorage-controller"
+const (
+	controllerAgentName = "etcdproxy-controller"
+
+	epcNamespace = "kube-apiserver-storage"
```

Comment:
This is fine for a backup.  I recommend using a construct like this for a primary: https://github.com/openshift/library-go/blob/master/pkg/config/leaderelection/leaderelection.go#L91-L95

```go
	if len(ret.Namespace) == 0 {
		if len(defaultNamespace) > 0 {
			ret.Namespace = defaultNamespace
		} else {
			// Fall back to the namespace associated with the service account token, if available
			if data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
				if ns := strings.TrimSpace(string(data)); len(ns) > 0 {
					ret.Namespace = ns
				}
			}
		}
	}
```